lib/HalModuleParser.js

```diff
@@ -564,25 +666,17 @@ HalModuleParser.prototype = {
 		const extensions = [];
 		if (suffixSize > MIN_MODULE_SUFFIX_SIZE + 4) {
 			idx = fileBuffer.length - 4 - suffixSize;
+
 			while (idx > 0 && idx < fileBuffer.length - 4) {
-				const type = fileBuffer.readUInt16LE(idx);
-				idx += 2;
-				const length = fileBuffer.readUInt16LE(idx);
-				idx += 2;
-				if (type === ModuleInfo.ModuleInfoExtension.END || type === ModuleInfo.ModuleInfoExtension.INVALID) {
+				const ext = this._readExtension(fileBuffer, idx);
```

Comment:
@monkbroc The need for extensions in prefix unfortunately came as a requirement for RTL872x-based platforms, because applications grow left on those platforms (end address is fixed, begin address moves left): https://github.com/particle-iot-inc/device-constants/pull/32

Binary version reader is pretty low level, hence the current implementation, but I agree it would be convenient to have a common `assets` array exposed. The problem is that you can separately parse prefix/suffix, so the only way we can probably do this sanely is with `parseFile` API.

Reply:
We can add `assets` array to both suffixInfo and prefixInfo objects to simplify parsing though (in addition to extensions list as there are other things in there not only asset dependencies).